ister_gui.py

```diff
@@ -350,8 +376,20 @@ def search_swap(choices, config, mount_d):
                 output = output[idx + len(pttr):]
                 output = output[:output.index('"')]
                 if output == 'swap':
-                    disk = ''.join(x for x in part if not x.isdigit())
+                    # first try to set using lsblk -no pkname part
+                    disk = os.path.join('/dev', get_part_devname(part))
+                    # if that fails, fallback to old (unreliable) method
```

Comment:
Has failure been something that has happened? I'm inclined to just say we have a bug in the lsblk toolchain path if we ever see it fail rather than potentially introduce bad configuration data.

Reply:
No, failure hasn't happened, but the old method works most of the time.

How about reporting an Alert() when the first one fails that the install may not succeed?

Reply:
I suspect if the first one fails the second one is going to have the same/worse problems (wrong path and something is funky with hardware for instance). But if this seems more likely to succeed leave it in.

Reply:
My thought here was to include the fallback in the case you mentioned where we have a bug in the lsblk toolchain. In that case, the lsblk command might fail but we can still detect partition info using the old method.

Reply:
Alrighty.